adapters/jsdom-svelte/src/index.ts

```diff
@@ -0,0 +1,215 @@
+
+    const elem = async () => {
+        const container = typeof containerOrFn === 'function' ? containerOrFn() : containerOrFn;
+        if (!container) {
+            throw new Error('Svelte base driver - element was not found');
+        }
+        return container;
+    };
+
+    const exists = async () => {
+        try {
+            await elem();
+            return true;
+        } catch (e) {
+            if (isMultipleElementsWithLocatorError(e)) {
+                throw e;
+            } else {
+                return false;
+            }
+        }
+    };
+
+    const handleCheckableInput = async (input: HTMLInputElement) => {
+        input.checked = !input.checked;
+        await fireEvent.input(input);
+        await fireEvent.change(input);
+    }
+
+    return {
+        $: (loc: Locator) => {
+            const getElement = async () => {
+                const container = await elem();
+                const elements = container.querySelectorAll(loc);
+                if (!elements.length) {
+                    throw new NoElementWithLocatorError(loc);
+                } else if (elements.length > 1) {
+                    throw new MultipleElementsWithLocatorError(elements.length, loc);
+                }
+                return elements[0];
+            };
+            return jsdomSvelteUniDriver(getElement);
+        },
+        $$: (selector: Locator) => jsdomSvelteUniDriverList(async () => {
+            const e = await elem();
+            return Array.from(e.querySelectorAll(selector));
+        }),
+        text: async () => elem().then((e) => e.textContent || ''),
+        value: async () => {
+            const e = (await elem()) as HTMLInputElement;
+            return e.value;
+        },
+        click: async () => {
+            const el = await elem();
+            await fireEvent.mouseDown(el);
+
+            if (elementIsFocusable(el)) {
+                if (document.activeElement != el) {
+                    if (document.activeElement) {
+                        fireEvent.blur(document.activeElement);
+                    }
+
+                    if (!el.disabled) {
+                        el.focus();
+                        await fireEvent.focus(el);
+                    }
+                }
+            }
+
+
+            await fireEvent.mouseUp(el);
+            await fireEvent.click(el);
+
+            if (isCheckable(el)) {
+                handleCheckableInput(el as HTMLInputElement);
+            }
+        },
+        mouse: {
+            press: async() => {
+                const el = await elem();
+
+                await fireEvent.mouseDown(el);
+            },
+            release: async () => {
+                const el = await elem();
+
+                await fireEvent.mouseUp(el);
+            },
+            moveTo: async (to) => {
+                const el = await elem();
+                const {left, top} = (await to.getNative()).getBoundingClientRect();
```

Comment:
Will this work in all testing environments?

Reply:
Mostly this is re-implementation of the react driver, so if it works there it should work here as well. Not sure we tested all possible environments though

Reply:
wdym by testing environment? jsdom / puppeteer/ etc?
it's only relevant to jsdom-svelte, on others you will use their own adapters

Reply:
Yeah you are right - sorry for that